scripts/run_tests.sh

```diff
@@ -6,18 +6,31 @@ set -e
 (cd /app/relation_engine_server/test/spec_release && \
   tar czvf spec.tar.gz sample_spec_release)
 
-black .
-flake8 --max-complexity 20 /app
-mypy --ignore-missing-imports /app
-bandit -r /app
+# Run code quality checks before tests; a failure here
+# will prevent the tests from running.
+source_dirs="/app/relation_engine_server /app/client_src /app/spec"
+flake8 --max-complexity 20 $source_dirs
+mypy --ignore-missing-imports $source_dirs
+bandit -r $source_dirs
```

Comment:
can you add in the black code formatting check here?

Reply:
done, also added as make task for running manually w/o having to invoke tests

Reply:
I do think it would be best if black during tests would run in read-only mode and then have the dev do black formatting manually. I did not see a black option to do so. Of course, one could do something funky like copy the source code, run black, and inspect for modified files.

Reply:
I'm sure there is one as I remember mentioning it to you... let me see if I can find it.

Reply:
yes: https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#exit-code